pandas/core/generic.py

```diff
@@ -5825,14 +5825,22 @@ def astype(
                         "Only a column name can be used for the "
                         "key in a dtype mappings argument."
                     )
+
+            # GH#44417 cast to Series so we can use .iat below, which will be
+            #  robust in case we
+            from pandas import Series
+
+            dtype = Series(dtype, dtype=object)
```

Comment:
i am surprised this is not a typing error. can you use something besides `dtype` here, maybe `result` or something.